### DISCUSSION TITLE
Problem with mouse and First Person Camera on Ubuntu 22 WSL2

### DISCUSSION THREAD

**Discussion by Grave18** [raysan5/raylib#2904] [Q&A]
3d camera_first_person example from [https://www.raylib.com/examples.html](url)

When I run this example the cursor is not blocked and because of this, the first-person camera behaves unpredictably. As it should be: On my Ubuntu laptop, when I start this example cursor disappears and the looking around works fine. On Ubuntu on WSL2, the cursor remains active and this is aproblem.

**Comment by chriscamacho** [comment]
you can very easily use WSL2 to compile native windows executables.
for an example that worked with WSL and should still be pertinent see
https://bedroomcoders.co.uk/looking-again-at-compiling-a-windows-raylib-app/

**Reply by Grave18** [reply]
Thanks for the offer. Apparently WSL2 is not very suitable for Raylib

**Reply by chriscamacho** [reply]
if you want to run something on windows create a windows executable, similarly for Linux, meet the OS half way

If you want truly cross platform "executable" then use web assembly
https://github.com/raysan5/raylib/wiki/Working-for-Web-(HTML5)
